ListInterface.h

```diff
@@ -0,0 +1,47 @@
+/*
+<Author>: Tucker J. Polomik github.com/TheNergaL
+<Name of algorithm>: Singly Linked-list
+<Brief description>:

```

Comment:
Change the "description comment" as mentioned for the other file. You can leave the complexities.